### DISCUSSION TITLE
Cyclic dependencies

### DISCUSSION THREAD

**Discussion by WildOrangutan** [LemonAppDev/konsist#739] [Ask a Question]
Is there an easy way to check cyclic dependencies, like in [ArchUnit](https://www.archunit.org/use-cases)?

**Comment by igorwojda** [comment]
By default, Konsist will not allow cyclic dependencies.
If you think this is broken then please post some examples (the sample project would be the best).

BTW I am planning to focus more on architectural checks after the upcoming `0.14.0` release (in `0.15.0`), so sample projects would help us speed development and match community expectations.

**Reply by WildOrangutan** [reply]
> By default, Konsist will not allow cyclic dependencies.

I don't understand. With what code do you trigger that assertion then?

> If you think this is broken then please post some examples (the sample project would be the best).

No. Nothing is broken. Only checking out the library.

**Reply by igorwojda** [reply]
The `assertArchitecture` is a great starting point:
Take a look at this https://docs.konsist.lemonappdev.com/writing-tests/architecture-assert

For now, checks for circular dependencies are not configurable (I will take another look it soon)
If you have a particular check in mind please let me know.

**Reply by igorwojda** [reply]
Also take a look at our internal tests (the have a nice mermaid diagrams)
https://github.com/LemonAppDev/konsist/tree/main/lib/src/apiTest/kotlin/com/lemonappdev/konsist/architecture/dependencyrules

**Comment by ogesaku** [comment]
@igorwojda  Is there an option to assert generic no cyclic dependency rule between all packages like in [ArchUnit](https://www.archunit.org/userguide/html/000_Index.html#_cycle_checks)?

```
slices().matching("com.myapp.(*)..").should().beFreeOfCycles()
```
